src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,25 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserRegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user != null) {
+            if (user.getLogin() != null && user.getLogin().length() >= 6) {
+                if (user.getPassword() != null && user.getPassword().length() >= 6) {
+                    if (user.getAge() != null && user.getAge() >= 18) {
+                        if (storageDao.get(user.getLogin()) == null) {
+                            return storageDao.add(user);
+                        }
+                    }
+                }
+            }
+        }
+        throw new UserRegistrationException();
```

Comment:
The `UserRegistrationException` is thrown without a message. It's a good practice to provide a meaningful message to indicate why the registration failed, which can help with debugging and user feedback.